File: linked-list/palindrome-go/solution_v2.go
package palindrome

func isPalindromeV2(node *Node) bool {
	fast := node
	slow := node
	var stack []rune

	for fast != nil && fast.Next != nil {
		stack = append(stack, slow.Letter)
		slow = slow.Next
		fast = fast.Next.Next
	}
	if fast != nil {
		slow = slow.Next
	}
	for slow != nil {
		value := stack[len(stack)-1]
		stack = stack[:len(stack)-1]

		if value != slow.Letter {
			return false
		}
		slow = slow.Next
	}

	return true
}

File: linked-list/loop-detection-java/src/main/java/SolutionV2.java
public class SolutionV2 {
    public Character loopDetectionV2(Node node) {
        Node fast = node.next.next;
        Node slow = node;

        while (true){
            if (fast.next == null || fast.next.next == null || slow == null){
                return  null;
            }

            if (fast == slow){
                return fast.next.data;
            }

            fast = fast.next.next;
            slow = slow.next;
        }
    }
}
